### ISSUE TITLE
[Circle]: Streamlined preferences

### ISSUE THREAD

**Issue by bragefuglseth** [SoongNoonien/plattenalbum#133]
Breakout thread from the ongoing [GNOME Circle review](https://gitlab.gnome.org/Teams/Circle/-/issues/219) of the app.

The app currently has some preferences that seem weird/unnecessary to have. In general, one should attempt to [choose their preferences](https://omter.com/preferences.html) carefully to make the actually useful preferences easier to locate, and to easen the maintenance burden. The preferences don't necessarily need to be *removed* altogether, but there is often a more practical way to expose a given piece of functionality. This will in turn make the app easier to use. Win-win!

The connection settings can maybe be dropped in favor of a "Disconnect" button in the main menu that takes you to a guided setup for a new connection. See issue #

- **Show stop button** — What makes the stop button useful? It's not something most modern music players have, as the pause button basically accomplishes the same thing from a user perspective. I'm not sure it's needed at all.
- **Show audio format** — These kinds of techincal details aren't really relevant when one just wants to play their music. If they are technically useful in some cases (such as when streaming from a remote server), one can put them into a popup directly in the player bar instead of having a global show/hide preference.
- **Support “MPRIS”** – This should just always be on, there's no good reason to have a way to disable it.
- **Sort albums by Year** – This could be exposed in a popover in the actual album list instead of hidden in the preferences list. Have a look at how Files is doing it.
- **Rewind via Previous button** — This is common behavior for all music players, no need to have a preference for it
- **Stop playback on quit** – Not entirely sure what this does under the hood, but would it make sense to rename it to "Run in Background"?

**Comment by SoongNoonien** [comment]
> Show stop button — What makes the stop button useful? It's not something most modern music players have, as the pause button basically accomplishes the same thing from a user perspective. I'm not sure it's needed at all.

Yes, this could be removed. I don't use the stop button myself.

> Show audio format — These kinds of techincal details aren't really relevant when one just wants to play their music. If they are technically useful in some cases (such as when streaming from a remote server), one can put them into a popup directly in the player bar instead of having a global show/hide preference.

I use them all the time and really don't want to remove them. But disabling them by default could be an option.

> Support “MPRIS” – This should just always be on, there's no good reason to have a way to disable it.

I'd say so too, but the situation with MPD is a bit different in that regard. See #105.

> Sort albums by Year – This could be exposed in a popover in the actual album list instead of being hidden in the preferences list.

Yes, this could be done that way, though I think this is nothing someone needs to adjust all the time. I think this is a preference to set and forget.

> Rewind via Previous button — This is common behavior for all music players, no need to have a preference for it.

The default (and in fact only possible) behavior for MPD is to jump to the last song rather than to jump to the beginning of the current song. But some people don't like this so I've implemented this. I myself use and really like the default behavior of MPD.

> Stop playback on quit – Not entirely sure what this does under the hood, but would it make sense to rename it to "Run in Background"?

If we were to rename it to "Run in Background" then we would also need to invert the toggle. Currently if "Stop playback on quit" is active the player issues a stop command to MPD when it is closed. So this is effectively the same as pressing the stop button right before closing the player. Perhaps this isn't actually needed but most MPD clients provide something similar.

**Comment by bragefuglseth** [comment]
Thanks a lot for the context!

> I'd say so too, but the situation with MPD is a bit different in that regard. See https://github.com/SoongNoonien/plattenalbum/issues/105.

That makes sense. Maybe it could be reworded to something like this:

```
Title: Send Playback to System
Subtitle: Enable if the MPD server doesn't communicate with MPRIS on its own.
```

> I use [audio format] all the time and really don't want to remove them. But disabling them by default could be an option.

Long-term it could maybe be presented differently in the app without needing a preference to be enabled, but this is a good initial solution.

> Yes, this could be done that way, though I think [album sorting] is nothing someone needs to adjust all the time. I think this is a preference to set and forget.

In that case I'd maybe reconsider if it's worth having at all :smile:

But I'd at least make it clear how the albums are sorted if they're *not* sorted by year. Is it alphabetically? If so, a "Sorting" dropdown with "Alphabetically" and "By Year" as alternatives would make sense.

> If we were to rename it to "Run in Background" then we would also need to invert the toggle. Currently if "Stop playback on quit" is active the player issues a stop command to MPD when it is closed.

I see, so it applies to MPD and not the app itself. Maybe it could be reworded to

```
Title: Run daemon in background
Subtitle: Keep MPD running after closing the app.
```

**Comment by SoongNoonien** [comment]
> I see, so it applies to MPD and not the app itself. Maybe it could be reworded to
Title: Run daemon in background
Subtitle: Keep MPD running after closing the app

Hm, I don't know if you understood me correctly. MPD will be left running anyways just the playback will be stopped.

**Comment by bragefuglseth** [comment]
Ah, my bad. It can probably be left as-is, then.

**Comment by bragefuglseth** [comment]
Maybe with the subtitle "Send a stop signal to MPD upon closing the app"? That would add some useful context.

**Comment by SoongNoonien** [comment]
I've now removed the stop button and sorting settings and changed the default for the audio format.

**Comment by SoongNoonien** [comment]
Regarding the MPRIS setting. I don't think "Send Playback to System" is understandable. When I read something like that I'm imaging some kind of audio stream being sent somewhere. Also I think "Enable if the MPD server doesn't communicate with MPRIS on its own" is not correct. The server itself never communicates on its own via the MPRIS protocol. It's always another client which provides an MPRIS interface.

**Comment by bragefuglseth** [comment]
> I'm not sure what exactly is not clear with current label. Is it confusing what "on quit" means?

No, but the fact that the signal is sent to MPD is probably good to have in there. Might not be obvious to everyone.

> Regarding the MPRIS setting. I don't think "Send Playback to System" is understandable. When I read something like that I'm imaging some kind of audio stream being sent somewhere. Also I think "Enable if the MPD server doesn't communicate with MPRIS on its own" is not correct. The server itself never communicates on its own via the MPRIS protocol. It's always another client which provides an MPRIS interface.

I see. Given that the setting is only useful to people who most likely know what MPRIS is already, there's probably no harm in keeping the technical wording, but it would be good to have a subtitle explaining who the preference is for / why it's useful nonetheless. You're better equipped for coming up with an accurate description of the setting than I am, but maybe something like "Enable if MPRIS integration is not provided by another client"? Use whatever you think is the most descriptive. The goal is really to avoid confusion about what the preference is about / why it exists. People who don't need the preference should also understand that they don't need to touch it.

Plattenalbum more technical in nature than the average music playing app, so I understand that a lot of its labels and descriptions need to remain technical as well. There's still room for adding context to them, though, for those of us who are new to MPD as a whole :grin:

**Comment by bragefuglseth** [comment]
Oh, and by the way, maybe "Show Audio Format" should be renamed to "Show Audio Details"? Considering that it shows kb/s as well.

**Comment by SoongNoonien** [comment]
I've added "Disable if “MPRIS” is supported by another client" as a subtitle to the MPRIS setting.

> If someone who doesn't know what MPRIS is at all stumbles over it and accidentally enables it, they may end up confused about why the currently played song suddenly isn't shown in the notifications area.

I think this shouldn't happen since MPRIS is enabled by default. So they would need to disable it first. And the expected reason for disabling should now be clear.



>>    I'm not sure what exactly is not clear with current label. Is it confusing what "on quit" means?

>No, but the fact that the signal is sent to MPD is probably good to have in there. Might not be obvious to everyone.

I'm afraid I still struggle with this. I thought it was "obvious" that the music is played by MPD and the client just controls the server therefore "stop playback" implies a communication with the server. Otherwise things like the play button should also have a tooltip saying "Send play signal to MPD" and not simply "Play". Which I think we can both agree on is a bit ridiculous 😁

> Oh, and by the way, maybe "Show Audio Format" should be renamed to "Show Audio Details"? Considering that it shows kb/s as well.

This is difficult. For me "Show Audio Details" somehow implies more details like a complete table of tags or even encoder settings. While "Audio Format" should also cover the (overall) bitrate and the bitrate mode (variable or constant) which both can be read of the current bitrate. At least MPD specifies it like that in the [docs](https://mpd.readthedocs.io/en/latest/user.html#audio-output-format). Also I tend to be cautious with string changes for translation reasons.

**Comment by bragefuglseth** [comment]
> I've added "Disable if “MPRIS” is supported by another client" as a subtitle to the MPRIS setting.

Sounds good to me! :+1:

> This is difficult. For me "Show Audio Details" somehow implies more details like a complete table of tags or even encoder settings. While "Audio Format" should also cover the (overall) bitrate and the bitrate mode (variable or constant) which both can be read of the current bitrate.

I think that makes sense; it can be kept then.

> I'm afraid I still struggle with this. I thought it was "obvious" that the music is played by MPD and the client just controls the server therefore "stop playback" implies a communication with the server.

Makes sense as well. I guess the technical implications of the preference are only relevant for people who have a more "advanced" MPD setup anyways.

Is there anything left to address from the list now? If not, let's close this.

**Comment by SoongNoonien** [comment]
> Is there anything left to address from the list now?

No, I don't think so. This is how it looks now:
![Bildschirmfoto vom 2024-07-18 23-07-49](https://github.com/user-attachments/assets/30b5b88b-0bea-471b-8497-93a3219fdeed)

**Comment by bragefuglseth** [comment]
Looks good! I'll tick of the box in the review.
